src/ScriptCs.Hosting/ScriptServicesBuilder.cs

```diff
@@ -48,7 +49,7 @@ public ScriptServices Build()
 
             _scriptEngineType = (Type)Overrides[typeof(IScriptEngine)];
 
-            var initDirectoryCatalog = _scriptName != null || _repl;
+            var initDirectoryCatalog = _scriptName != null || _repl || _loadScriptPacks;
```

Comment:
this is still kinda confusing since I can say `LoadScriptPacks(false)` and yet it will override me because I have set a filename. It would be better to just check `_loadScriptPacks` here - this way I can have script execution or REPL without initializing script packs if I don't care about them in my host